phpBB/includes/acp/acp_update.php

```diff
@@ -52,6 +54,102 @@ function main($id, $mode)
 		}
 
+		if ($request->is_set('1clickupdate'))
+		{
+			if (empty($updates_available))
+			{
+				trigger_error($user->lang('NO_UPDATES_AVAILABLE') . adm_back_link($this->u_action));
+			}
+
+			if (confirm_box(true))
+			{
+				if (empty($updates_available['download']) || empty($updates_available['signature']))
+				{
+					trigger_error($user->lang('NO_1CLICKUPDATE') . adm_back_link($this->u_action));
+				}
+
+				$filesystem = new \Symfony\Component\Filesystem\Filesystem();
+
+				// generate local filename for new version
+				$tmp_filename = 'phpbb-update-' . $updates_available['current'] . '.zip';
+
+				// generate path where new version will be temporarily stored and make sure it doesn't exist
+				$tmp_path = $phpbb_root_path . 'store/' . $tmp_filename;
+				if (file_exists($tmp_path))
+				{
+					trigger_error($user->lang('1CLICKUPDATE_TMP_FILE_EXISTS') . adm_back_link($this->u_action));
+				}
+
+				// download new version
+				$client = new \GuzzleHttp\Client;
+				$client->request('GET', $updates_available['download'], ['sink' => $tmp_path]);
+				
+				// download signature of the new version
+				$signature_response = $client->request('GET', $updates_available['signature']);
+				$signature = $signature_response->getBody();
+
+				// verify integrity of the downloaded file
+				$rsa = new RSA();
+				$key = file_get_contents($phpbb_root_path . 'phpbb/update/key/key');
+				$rsa->loadKey($key, RSA::PRIVATE_FORMAT_PKCS1);
+				$rsa->setSignatureMode(RSA::SIGNATURE_PKCS1);
+				$hash = sha1_file($tmp_path);
+				if (!$rsa->verify($hash, $signature))
+				{
+					// remove file
+					$filesystem->remove($tmp_path);
+
+					trigger_error($user->lang('SIGNATURE_MISMATCH') . adm_back_link($this->u_action));
+				}
+				else
+				{
+					// unzip new version
+					$zip = new \ZipArchive;
+					$res = $zip->open($tmp_path);
+					if ($res !== true)
+					{
+						// remove file
+						$filesystem->remove($tmp_path);
+
+						trigger_error($user->lang('UNZIP_FAILED') . adm_back_link($this->u_action));
+					}
+					
+					$tmp_path_dir = substr($tmp_path, 0, -3);
+					$zip->extractTo($tmp_path_dir);
+					$zip->close();
+
+					// remove config, images, store and foles from new version
+					$filesystem->remove($tmp_path_dir . '/phpBB3/config.php');
+					$filesystem->remove($tmp_path_dir . '/phpBB3/images');
+					$filesystem->remove($tmp_path_dir . '/phpBB3/store');
+					$filesystem->remove($tmp_path_dir . '/phpBB3/files');
+
+					// remove vendor and cache from current version
+					$filesystem->remove($phpbb_root_path . 'vendor');
+					$filesystem->remove($phpbb_root_path . 'cache');
+
+					// copy new version to root path
+					$filesystem->mirror($tmp_path_dir . '/phpBB3/', $phpbb_root_path);
+
+					// remove temporary files
+					$filesystem->remove($tmp_path);
+					$filesystem->remove($tmp_path_dir);
+
+					// redirect user to installation script
+					redirect(append_sid($phpbb_root_path . 'install/app.' . $phpEx));
```

Comment:
So you'll be redirecting admins to run the database update. Shouldn't it be possible to just run migrations from here?

Reply:
It is of course possible, but it's an extra tons of work because we need to be careful with timeouts, permissions, etc, everything correctly handled by the original installer. I'm not sure if it's worth it...